app/models/reducer.rb

```diff
@@ -52,6 +52,7 @@ def self.of_type(type)
 
   def process(extract_fetcher, reduction_fetcher, relevant_reductions=[])
     light = Stoplight("reducer-#{id}") do
+      extract_fetcher.strategy! (if running_reduction? then :fetch_minimal else :fetch_all end)
```

Comment:
perhaps use a ternary with variable assignment for readability? 

Also the `extract_fetcher` class has the attr_accessor for variables including `strategy`, if you aren't chaining the `extract_fetcher` then to me the code is clearer using normal variable assignment. That way i don't have to lookup / remember what `strategy!` does.
```suggestion
      fetcher_strategy = running_reduction?  ? :fetch_minimal : :fetch_all
      extract_fetcher.strategy = fetcher_strategy
```